Vostok.ClusterClient.Transport.Tests/Functional/Common/SendFailureTests.cs

```diff
@@ -0,0 +1,53 @@
+using System.Net.Sockets;
+using System.Threading;
+using FluentAssertions;
+using NUnit.Framework;
+using Vostok.Clusterclient.Core.Model;
+using Vostok.Clusterclient.Transport.Tests.Helpers;
+using Vostok.Commons.Environment;
+using Vostok.Commons.Threading;
+
+namespace Vostok.Clusterclient.Transport.Tests.Functional.Common
+{
+    internal abstract class SendFailureTests<TConfig> : TransportFunctionalTests<TConfig>
+        where TConfig : ITransportTestConfig, new()
+    {
+        [Test]
+        public void Should_return_SendFailure_code_when_socket_closed_while_sending_large_body()
+        {
+            if (!RuntimeDetector.IsDotNetCore21AndNewer)
+            {
+                return;
+            }
+            
+            using (var autoResetEvent = new AutoResetEvent(false))
```

Comment:
How about hiding `autoResetEvent` inside `BlockingStream`?

I think it will become easier to reuse this combination for future users